vm/src/vm.rs

```diff
@@ -573,7 +572,7 @@ impl VirtualMachine {
                     &bytecode::Constant::Code { ref code } => {
                         PyObject::new(PyObjectKind::Code { code: code.clone() }, self.get_type())
                     }
-                    &bytecode::Constant::None => PyObject::new(PyObjectKind::None, self.get_type()),
+                    &bytecode::Constant::None => self.ctx.none.clone(),
```

Comment:
Am I right in thinking that we're just cloning the reference here (so there is still just one None)?

Reply:
Yes, that's the plan.

There should only be one place where we create a new PyObjectKind::None, and that's when we create the context. (There's also a temporary none created as a placeholder while we construct the circular reference in type_type, but it shouldn't escape.)

I don't know of a way to prevent accidental creation of more Nones later.